test/Microsoft.AspNet.Mvc.FunctionalTests/ConsumesAttributeTests.cs

```diff
@@ -49,11 +49,11 @@ public ConsumesAttributeTests(MvcTestFixture<BasicWebSite.Startup> fixture)
 
             // Act
             var response = await Client.SendAsync(request);
-            var product = JsonConvert.DeserializeObject<Product>(await response.Content.ReadAsStringAsync());
+            var body = await response.Content.ReadAsStringAsync();
 
             // Assert
-            Assert.Equal(HttpStatusCode.NoContent, response.StatusCode);
-            Assert.Null(product);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("ConsumesAttribute_PassThrough_Product_Json", body);
```

Comment:
The tests used to receive the product in the body and just return null, which produced 204. With this change, the tests returned 415, so we removed the [FromBody] from the controllers (so that action selection happens normally) and changed the response so that we can verify the right action was hit.